src/ui/pages/WalletSelect/WalletSelect.tsx

```diff
@@ -17,7 +17,61 @@ import { Button } from 'src/ui/ui-kit/Button';
 import { VStack } from 'src/ui/ui-kit/VStack';
 import { Background } from 'src/ui/components/Background';
 import { Spacer } from 'src/ui/ui-kit/Spacer';
+import type { WalletGroup } from 'src/shared/types/WalletGroup';
+import { isReadonlyContainer } from 'src/shared/types/validators';
+import { getAddressesPortfolio } from 'src/shared/analytics/shared/getTotalWalletsBalance';
+import { useCurrency } from 'src/modules/currency/useCurrency';
+import { NeutralDecimals } from 'src/ui/ui-kit/NeutralDecimals';
+import { formatCurrencyToParts } from 'src/shared/units/formatCurrencyValue';
+import PortfolioIcon from 'jsx:src/ui/assets/portfolio.svg';
+import { Media } from 'src/ui/ui-kit/Media';
 import { WalletList } from './WalletList';
+import * as styles from './styles.module.css';
+
+function PortfolioRow({ walletGroups }: { walletGroups: WalletGroup[] }) {
+  const { currency } = useCurrency();
+
+  const addresses = useMemo(() => {
+    return walletGroups
+      .filter((group) => !isReadonlyContainer(group.walletContainer))
+      .flatMap((group) =>
+        group.walletContainer.wallets.map((wallet) => wallet.address)
+      );
+  }, [walletGroups]);
+
+  const { data: portfolio, isLoading } = useQuery({
+    queryKey: ['getAddressesPortfolio', addresses, currency],
+    queryFn: () => getAddressesPortfolio({ addresses, currency }),
```

Comment:
I think, we should use a usual definition-sdk hook here. There is probably a predefined one in the library.